packages/react-native-mmkv/src/useMMKVDevTools.ts

```diff
@@ -1,104 +1,105 @@
- */
-export function useMMKVDevTools({
-  errorHandler,
-  storage = new MMKV(),
-}: {
+type Params = {
   errorHandler?: (error: Error) => void;
   storage?: MMKV;
-} = {}) {
+};
+
+export function useMMKVDevTools({
+  errorHandler,
+  storage: storageProp,
+}: Params = {}) {
   const client = useDevToolsPluginClient('mmkv');
 
+  // Ensure we create MMKV only once if not provided
+  const storage = useMemo(() => storageProp ?? createMMKV(), [storageProp]);
+
   const handleError = useCallback(
     (error: unknown) => {
-      if (error instanceof Error) {
-        errorHandler?.(error);
-      } else {
-        errorHandler?.(new Error(`Unknown error: ${String(error)}`));
-      }
+      const err = error instanceof Error ? error : new Error(String(error));
+      errorHandler?.(err);
     },
     [errorHandler]
   );
 
   useEffect(() => {
+    if (!client) return;
+
     const on = (
       event: Method,
       listener: (params: { key?: string; value?: string }) => Promise<any>
-    ) =>
-      client?.addMessageListener(event, async (params: { key?: string; value?: string }) => {
+    ): EventSubscription =>
+      client.addMessageListener(event, async (params: { key?: string; value?: string }) => {
         try {
           const result = await listener(params);
-
-          client?.sendMessage(`ack:${event}`, { result });
+          client.sendMessage(`ack:${event}`, { result: result ?? true });
         } catch (error) {
+          // send a serializable error payload
+          const err = error instanceof Error ? { message: error.message, stack: error.stack } : { message: String(error) };
           try {
-            client?.sendMessage('error', { error });
+            client.sendMessage('error', err);
+          } finally {
             handleError(error);
-          } catch (e) {
-            handleError(e);
           }
         }
       });
 
-    const subscriptions: (EventSubscription | undefined)[] = [];
+    const subscriptions: EventSubscription[] = [];
 
-    try {
-      subscriptions.push(
-        on('getAll', async () => {
-          const keys = storage.getAllKeys();
-          return keys?.map((key) => [key, storage.getString(key)]);
-        })
-      );
-    } catch (e) {
-      handleError(e);
-    }
+    // getAll
+    subscriptions.push(
+      on('getAll', async () => {
+        const keys = storage.getAllKeys() ?? [];
+        // Try to read strings; fall back to number/bool if not string
+        return keys.map((key) => {
+          const s = storage.getString(key);
+          if (s !== undefined) return [key, s];
 
-    try {
-      subscriptions.push(
-        on('set', async ({ key, value }) => {
-          if (key !== undefined && value !== undefined) {
-            return storage.set(key, value);
-          }
-        })
-      );
-    } catch (e) {
-      handleError(e);
-    }
+          const n = storage.getNumber?.(key);
```

Comment:
```suggestion
          const n = storage.getNumber(key);
```

no need for optional here and below?